exp/api/v1beta2/rosaroleconfig_types.go

```diff
@@ -0,0 +1,158 @@
+	Region             string                        `json:"region,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=rosaroleconfigs,scope=Namespaced,categories=cluster-api,shortName=rosarole
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+
+// ROSARoleConfig is the Schema for the rosaroleconfigs API
+type ROSARoleConfig struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ROSARoleConfigSpec   `json:"spec,omitempty"`
+	Status ROSARoleConfigStatus `json:"status,omitempty"`
+}
+
+type AccountRoleConfig struct {
+	// TODO: verify max len
+	// +kubebuilder:validation:MaxLength:=4
+	// +kubebuilder:validation:Required
+	Prefix string `json:"prefix"`
+	// +optional
+	PermissionsBoundaryARN string `json:"permissionsBoundaryARN,omitempty"`
+	// +optional
+	Path string `json:"path,omitempty"`
+	// +kubebuilder:validation:Required
+	Version string `json:"version"`
+	// +optional
+	SharedVPCConfig SharedVPCConfig `json:"sharedVPCConfig,omitempty"`
+}
+
+type OperatorRoleConfig struct {
+	// +kubebuilder:validation:MaxLength:=4
+	// +kubebuilder:validation:Required
+	Prefix string `json:"prefix"`
+	// +optional
+	PermissionsBoundaryARN string `json:"permissionsBoundaryARN,omitempty"`
+	// +kubebuilder:validation:Required
+	OIDCConfigID string `json:"oidcConfigId,omitempty"`
+	// +optional
+	SharedVPCConfig SharedVPCConfig `json:"sharedVPCConfig,omitempty"`
+}
+
+type SharedVPCConfig struct {
+	RouteRoleARN       string `json:"routeRoleARN,omitempty"`
+	VPCEndpointRoleARN string `json:"vpcEndpointRoleArn,omitempty"`
+}
+
+type OIDCConfig struct {
+	ManagedOIDC           bool                                      `json:"managedOIDC"`
+	ExternalAuthProviders []rosacontrolplanev1.ExternalAuthProvider `json:"externalAuthProviders,omitempty"`
+}
+
+type OIDCClient struct {
+	ComponentName      string                       `json:"componentName"`
+	ComponentNamespace string                       `json:"componentNamespace"`
+	ClientID           string                       `json:"clientID"`
+	ClientSecret       *corev1.LocalObjectReference `json:"clientSecret,omitempty"`
+}
+
+type SecretReference struct {
+	Name string `json:"name"`
+}
+
+// ROSARoleConfigStatus defines the observed state of ROSARoleConfig
+type ROSARoleConfigStatus struct {
+	OIDCID           string               `json:"oidcID,omitempty"`
+	OIDCProviderARN  string               `json:"oidcProviderARN,omitempty"`
+	AccountRolesRef  AccountRolesRef      `json:"accountRolesRef,omitempty"`
+	OperatorRolesRef OperatorRolesRef     `json:"operatorRolesRef,omitempty"`
+	Conditions       clusterv1.Conditions `json:"conditions,omitempty"`
+}
+
+type AccountRolesRef struct {
+	InstallerRoleARN string `json:"installerRoleARN,omitempty"`
+	SupportRoleARN   string `json:"supportRoleARN,omitempty"`
+	WorkerRoleARN    string `json:"workerRoleARN,omitempty"`
+}
+
+type OperatorRolesRef struct {
```

Comment:
may you can refer to the AWSRef struct [here](https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/controlplane/rosa/api/v1beta2/rosacontrolplane_types.go#L341) instead of creating new struct , what do you think ?